features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/startup/RumAppStartupDetector.kt

```diff
@@ -0,0 +1,38 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.rum.internal.startup
+
+import android.app.Application
+import com.datadog.android.core.InternalSdkCore
+import com.datadog.android.core.internal.system.BuildSdkVersionProvider
+import com.datadog.android.rum.DdRumContentProvider
+
+internal interface RumAppStartupDetector {
+    interface Listener {
+        fun onAppStartupDetected(scenario: RumStartupScenario)
+    }
+
+    fun destroy()
+
+    companion object {
+        fun create(
+            application: Application,
+            sdkCore: InternalSdkCore,
+            listener: Listener
+        ): RumAppStartupDetector {
+            val impl = RumAppStartupDetectorImpl(
```

Comment:
very nit: maybe just `return RumAppStartupDetectorImpl`, unless it is simple to step during debugging.

Or we can move even further: get rid of dedicated class and use `object: RumAppStartupDetector  { ... }`, like here https://github.com/DataDog/dd-sdk-android/blob/6e78b75d121be56a0b63ff660f10e27b82bf015e/features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/vitals/JankStatsProvider.kt#L25